### PR TITLE
feat(droneci): support for multiline string image dependency

### PR DESCRIPTION
**Pull request by ns-ggeorgiev**
Fix for issue #15576 

## Changes

The droneci manager is modified to properly detect when image is specified on multiple lines. For example:

    image: "amd64/node:10.0.0\
            @sha256:36adc17e9cceab32179d3314da9cb9c737ffb11f0de4e688f407ad6d9ca32201"

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ x ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ x ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ x ] Both unit tests + ran on a real repository: 
   https://github.com/ggeorgiev/renovate-test/pull/2/files
   https://github.com/ggeorgiev/renovate-test/pull/3/files

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
- please reduce debug logging to trace
- can you demonstrate the `.drone.yml` full updates on a public repo with the new code? (both regular style as well as the newly supported formats)

**Comment by ns-ggeorgiev** [comment]
@rarkins how I can do that. Even if I create the repo, how I can run the code from the fork on it?

**Comment by rarkins** [comment]
https://github.com/renovatebot/renovate/blob/main/docs/development/local-development.md#run-against-a-real-repo

File: lib/modules/manager/droneci/extract.ts [hunk 1]
```diff
@@ -3,26 +3,64 @@ import { newlineRegex, regEx } from '../../../util/regex';
 import { getDep } from '../dockerfile/extract';
 import type { PackageDependency, PackageFile } from '../types';
 
+export function getDroneDep(
+  currentFrom: string | null | undefined,
+  replaceString: string | undefined
+): PackageDependency {
+  const dep = getDep(currentFrom);
+  logger.debug(
```
Comment (rarkins) [inline lib/modules/manager/droneci/extract.ts:11]:
```suggestion
  logger.trace(
```

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs demo in real repo, both old and new cases to verify all works/updates

File: lib/modules/manager/droneci/extract.ts (hunk 1 above)
Comment (ns-ggeorgiev) [reply lib/modules/manager/droneci/extract.ts]: this one was debuf from before, I just moved the code in a helper method. Anyways, changed to trace

**Comment by ns-ggeorgiev** [comment]
real repo demo links in the added to the description

**Comment by ns-ggeorgiev** [comment]
@rarkins any reason to sit on this?

File: lib/modules/manager/droneci/extract.ts [hunk 2]
```diff
@@ -3,26 +3,64 @@ import { newlineRegex, regEx } from '../../../util/regex';
...
+): PackageDependency {
+  const dep = getDep(currentFrom);
+  logger.trace(
+    {
+      depName: dep.depName,
+      currentValue: dep.currentValue,
+      currentDigest: dep.currentDigest,
+    },
+    'DroneCI docker image'
+  );
```
Comment (viceice) [inline lib/modules/manager/droneci/extract.ts:18]:
```suggestion
```

File: lib/modules/manager/droneci/extract.ts [hunk 3]
```diff
@@ -3,26 +3,64 @@ import { newlineRegex, regEx } from '../../../util/regex';
...
+  dep.depType = 'docker';
+  dep.replaceString = replaceString;
```
Comment (viceice) [inline lib/modules/manager/droneci/extract.ts:20]: make optional and only override if passed

File: lib/modules/manager/droneci/extract.ts [hunk 4]
```diff
@@ -3,26 +3,64 @@ import { newlineRegex, regEx } from '../../../util/regex';
...
+              const dep = getDroneDep(currentFrom, replaceString);
+              deps.push(dep);
+            }
+            break;
+          }
+        }
+      } else {
+        const match = regEx(/^\s* image:\s*'?"?([^\s'"]+)'?"?\s*$/).exec(line);
+        if (match) {
+          const dep = getDroneDep(match[1], match[1]);
```
Comment (viceice) [inline lib/modules/manager/droneci/extract.ts:61]:
```suggestion
          const dep = getDroneDep(match[1]);
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**